x-pack/plugins/enterprise_search/server/routes/app_search/engines.test.ts

```diff
@@ -108,6 +108,73 @@ describe('engine routes', () => {
     });
   });
 
+  describe('POST /api/app_search/engines', () => {
+    const AUTH_HEADER = 'Basic 123';
+    const mockRequest = {
+      headers: {
+        authorization: AUTH_HEADER,
+      },
+      body: {
+        name: 'some-engine',
+        language: 'en',
+      },
+    };
+
+    let mockRouter: MockRouter;
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      mockRouter = new MockRouter({
+        method: 'post',
+        path: '/api/app_search/engines',
+        payload: 'body',
+      });
+
+      registerEnginesRoutes({
+        ...mockDependencies,
+        router: mockRouter.router,
+      });
+    });
+
+    it('creates a request handler', () => {
+      mockRouter.callRoute(mockRequest);
+
+      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
+        path: '/as/engines',
+        body: {
+          name: 'some-engine',
+          language: 'en',
+        },
+      });
+    });
+
+    it('passes the body to enterpriseSearchRequestHandler', () => {
+      mockRouter.callRoute({ body: { name: 'some-engine', language: 'en' } });
+
+      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
+        path: '/as/engines',
+        body: { name: 'some-engine', language: 'en' },
+      });
+    });
```

Comment:
oh you can delete this also, sorry! let me know if `enterpriseSearchRequestHandler` isn't super clear about what it can or can't take, I'm actually slightly surprised it wasn't throwing typescript errors/warnings for the unnecessary `body` key - I should probably look at my typings again